src/web/sitemap.rs

```diff
@@ -21,16 +39,30 @@ impl_webpage! {
 }
 
 pub fn sitemap_handler(req: &mut Request) -> IronResult<Response> {
+    let router = extension!(req, Router);
+    let which = cexpect!(req, router.find("which")).to_lowercase();
+
     let mut conn = extension!(req, Pool).get()?;
     let query = conn
         .query(
-            "SELECT DISTINCT ON (crates.name)
-                    crates.name,
-                    releases.release_time
+            "SELECT crates.name,
+                    MAX(releases.release_time) as release_time
              FROM crates
              INNER JOIN releases ON releases.crate_id = crates.id
-             WHERE rustdoc_status = true",
-            &[],
+             WHERE 
+                rustdoc_status = true AND 
+                ( 
+                    crates.name like $1 OR 
+                    crates.name like $2
```

Comment:
I think you can use ILIKE here to avoid needing two parameters.

Reply:
that was intentional, `ILIKE` only can use a `BTREE` index when the search doesn't start with non-alphabetic characters. I thought it was a little overkill to create a new index (`GIN` or `BTREE` on `lower(name)`) for that. 

(see also the [postgres docs on index types](https://www.postgresql.org/docs/13/indexes-types.html))

Another alternative would be to use the db proc `normalize_crate_name` , which already has a specific index that is used in `web::match_version`. But that also seemed a little off, but I'm happy to change it if you think that fits the project more. 

In any case I see this is definitely worth a comment :)